### DISCUSSION TITLE
Login no facebook com o Laravel Socialite não retorna parâmetros na bladeEscreva o titulo de forma clara e objetiva

### DISCUSSION THREAD

**Discussion by rhaymisonbetini** [frontendbr/forum#1612] [Perguntas e Respostas]
Usando socialite, posso retornar os dados do usuário do Facebook, mas não consigo jogar esses dados na minha rota: Abaixo está o código:

`` 
public function handleProviderCallback()
   {
        $user = Socialite::driver('facebook')->user();
        $user->token;
        return Socialite::driver('facebook')->with(['name' => $user->name ])->redirect();
    }
``
My Services.php:

``
'facebook' => [
        'client_id' => 'XXXXXXXXXXXXXXXXXXXXXXXxxx',
        'client_secret' => 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXx',
        'redirect' => 'http://localhost:8000/painel/cadastro/facebook',
    ],
``

Abaixo o erro

```
Undefined variable: name
```

Olhei a documentação e mostrava o meio que usei. Alguém saberia apontar ou instruir como resolver essa questão. Obrigado

laravel facebook

**Comment by HmmerHead** [comment]
> não consigo jogar esses dados na minha rota

Não entendi o que você quis dizer com isso. Se for na View, onde você deseja retornar esses dados, veja apenas se o a variável está lá, ou se não houve algum problema com na hora de transformar-lá em algum tipo de retorno.

Sendo algum outro tipo de coisa, informe exemplifique mais.

Corrija-me se eu estiver errado, mas esse tipo de erro me faz pensar que seja algo relacionado com a view.

**Comment by rhaymisonbetini** [comment]
> > não consigo jogar esses dados na minha rota
> 
> Não entendi o que você quis dizer com isso. Se for na View, onde você deseja retornar esses dados, veja apenas se o a variável está lá, ou se não houve algum problema com na hora de transformar-lá em algum tipo de retorno.
> 
> Sendo algum outro tipo de coisa, informe exemplifique mais.
> 
> Corrija-me se eu estiver errado, mas esse tipo de erro me faz pensar que seja algo relacionado com a view.

Na verdade essa variavel não esta sendo nem enviada pra View....... Tentoo recuperar ela como:
``
{{ $name }} 
``

O erro acontece agora....seu  eu uso "ou string vazia"
```
{{ $name ??  ' ' }}
```
Ela é mostrada normalmente..Então como o erro mostra, a variável name nem é enviada para View. no  With(). Agora o motivo eu não estou consindo descobrir

**Comment by HmmerHead** [comment]
Debugando, aparece alguma coisa? Com um dd() por exemplo. Dá algum retorno? Se não, pode ser a função.

Algo que percebi também, é que não há uma função que chama a view e adiciona um retorno.

Como mostra na documentação ( [Doc.](https://laravel.com/docs/5.8/views#passing-data-to-views) ), algo do tipo:

`return view('greetings', ['name' => 'Victoria']);`

Esse return está apenas devolvendo os dados, mas não há indicação de qual view seria essa.

**Comment by rhaymisonbetini** [comment]
O 



> Debugando, aparece alguma coisa? Com um dd() por exemplo. Dá algum retorno? Se não, pode ser a função.
> 
> Algo que percebi também, é que não há uma função que chama a view e adiciona um retorno.
> 
> Como mostra na documentação ( [Doc.](https://laravel.com/docs/5.8/views#passing-data-to-views) ), algo do tipo:
> 
> `return view('greetings', ['name' => 'Victoria']);`
> 
> Esse return está apenas devolvendo os dados, mas não há indicação de qual view seria essa.

O redirect para a view é feito justamente a função service.php que mostrei ali encima:
```
'facebook' => [ 'client_id' => 'XXXXXXXXXXXXXXXXXXXXXXXxxx', 'client_secret' => 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXx',
 'redirect' => 'http://localhost:8000/painel/cadastro/facebook', ], // aqui
``

**Comment by murilolivorato** [comment]
olá @rhaymisonbetini
escrevi este artigo sobre isso -
https://medium.com/@murilolivorato/login-with-facebook-in-a-vue-laravel-app-popup-redirect-oauth-04e4c077f6a4
veja se te ajuda

**Reply by rhaymisonbetini** [reply]
Oloko man.. isso ja tem 7 anos kkkkkkkkkkkkkkkk
